internal/pkg/levee/levee.go

```diff
@@ -75,11 +72,10 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
-func reportSourcesReachingSink(pass *analysis.Pass, sources []*source.Source, instr ssa.Instruction, propagations map[ssa.Node]propagation.Propagation) {
-	for _, s := range sources {
-		prop := propagations[s.Node]
+func reportSourcesReachingSink(pass *analysis.Pass, instr ssa.Instruction, propagations map[*source.Source]propagation.Propagation) {
+	for source, prop := range propagations {
 		if prop.HasPathTo(instr.(ssa.Node)) && !prop.IsSanitizedAt(instr) {
```

Comment:
I personally dont like the fact we need to do 2 function calls here. We should change this to 1 function call like:

prop.HasUnsanitizedPathTo(instr)?

Reply:
We think alike ;)

I actually made this exact comment on one of your recent PR's : https://github.com/google/go-flow-levee/pull/226#discussion_r544438336

I think this change is out of scope for this PR. I was planning to do it in a follow-up.

Reply:
PR here: #242